includes/deprecated/class-wc-order-compatibility-mondido.php

```diff
@@ -58,21 +58,9 @@ public function __construct( $the_order ) {
 		if ( FALSE === $the_order ) {
 			$the_order = $post;
 		} elseif ( is_numeric( $the_order ) ) {
-			if ( OrderUtil::custom_orders_table_usage_is_enabled() ) {
```

Comment:
This file will be removed after upgrade so no need to add any if checks.

Reply:
Is upgrading WooCommerce a prerequisite for using HPOS?

Reply:
Yes.

_October 2023
From WooCommerce 8.2, released on October 2023, High-Performance Order Storage (HPOS) is officially flagged as stable and will be enabled by default for new installations._

Reply:
I have tested everything on the latest version 8.7. So ideally if we get merchants to upgrade to the same one we should be safe. Hopefully the newer versions will also work, aka 9.0, but we should encourage our merchants to use 8.7 for now.

Reply:
What will happen if the merchant uses our updated plugin without updating their WooCommerce version? Do we really want to enforce all merchants to upgrade their WooCommerce version to use the latest version of our plugin?

Reply:
The idea is for us to be able to support HPOS and CPT at the same time, which is what this PR is about. Our merchants should till be able to run really old woo and it will work since it's backwards compatible. However, IMHO, i don't think it's good for them to be on old versions, like 2.2 etc. So if our plan is to ask those merchants to upgrade we might as well ask everyone. 

Merchants that are using versions 8.2 or later will be able to activate HPOS directly. However they will need to also do some admin work since they need to sync orders between the tables etc, it's an easy process but still should be monitored by them and if any errors should come up they would need to contact us or Woocommerce.